mlflow/pyfunc/model.py

```diff
@@ -228,17 +231,37 @@ def _save_model_with_class_artifacts_params(
     if artifacts:
         saved_artifacts_config = {}
         with TempDir() as tmp_artifacts_dir:
-            tmp_artifacts_config = {}
             saved_artifacts_dir_subpath = "artifacts"
             for artifact_name, artifact_uri in artifacts.items():
-                tmp_artifact_path = _download_artifact_from_uri(
-                    artifact_uri=artifact_uri, output_path=tmp_artifacts_dir.path()
-                )
-                tmp_artifacts_config[artifact_name] = tmp_artifact_path
-                saved_artifact_subpath = posixpath.join(
-                    saved_artifacts_dir_subpath,
-                    os.path.relpath(path=tmp_artifact_path, start=tmp_artifacts_dir.path()),
-                )
+                if artifact_uri.startswith("hf:/"):
+                    try:
+                        from huggingface_hub import snapshot_download
+
+                        repo_id = re.sub(r"^hf:/", "", artifact_uri)
+                        snapshot_location = snapshot_download(
+                            repo_id=repo_id,
+                            cache_dir=os.path.join(
+                                path, saved_artifacts_dir_subpath, artifact_name
+                            ),
+                        )
+                        saved_artifact_subpath = os.path.relpath(path=snapshot_location, start=path)
+                        if extra_pip_requirements is None:
+                            extra_pip_requirements = ["transformers"]
+                        elif "transformers" not in extra_pip_requirements:
+                            extra_pip_requirements.append("transformers")
+                    except Exception as e:
```

Comment:
Do we need this try-catch?

Reply:
It's just to make the exception message more clear I think

Reply:
What does the `snapshot_download`'s error message look like?

Reply:
If it's a invalid repo-id: `401 Client Error. (Request ID: Root=1-64e425ea-28542f9a441029ef45fca406)\n\nRepository Not Found for url: https://huggingface.co/api/models/invalid-repo-id/revision/main.\nPlease make sure you specified the correct `repo_id` and `repo_type`.\nIf you are trying to access a private or gated repo, make sure you are authenticated.\nInvalid username or password.'` But what if 'from huggingface import snapshot_download' gives ImportError, would it confuse customers?

Reply:
It would. `Failed to download snapshot from Hugging Face Hub`. I would throw a different error for `ImportError`.